### PR TITLE
Upgrade to Liquibase 4.24.0

### LINKED ISSUE #36446: Regression: Liquibase fails to migrate on Quarkus start, crashing the application
### Describe the bug

An application using Liquibase to set up the database or to migrate it when starting up crashes because of a Liquibase problem – looks like Liquibase tries to re-create its own changelog tables and fails.

The behaviour appeared after upgrading Quarkus, last working version was 3.3.3, upgrading to 3.4.x starts the issue to appear.

### Expected behavior

The application should behave like it did before upgrading Quarkus.

### Actual behavior

The application, when starting, terminates with such a stack trace:

```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2023-10-12 13:56:40,203 INFO  [liq.database] (main) Set default schema name to PUBLIC
2023-10-12 13:56:40,303 INFO  [liq.lockservice] (main) Successfully acquired change log lock
2023-10-12 13:56:40,542 INFO  [liq.changelog] (main) Creating database history table with name: PUBLIC.changelog
2023-10-12 13:56:40,549 INFO  [liq.changelog] (main) Reading from PUBLIC.changelog
No validation errors found.
2023-10-12 13:56:40,555 INFO  [liq.command] (main) Command execution complete
2023-10-12 13:56:40,582 INFO  [liq.changelog] (main) Creating database history table with name: PUBLIC.changelog
2023-10-12 13:56:40,584 INFO  [liq.command] (main) Command execution complete
2023-10-12 13:56:40,586 INFO  [liq.lockservice] (main) Successfully released change log lock
2023-10-12 13:56:40,639 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: java.lang.IllegalStateException: Error starting Liquibase
        at io.quarkus.liquibase.runtime.LiquibaseRecorder.doStartActions(LiquibaseRecorder.java:88)
        at io.quarkus.deployment.steps.LiquibaseProcessor$startLiquibase1744275855.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.LiquibaseProcessor$startLiquibase1744275855.deploy(Unknown Source)
        ... 11 more
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Table "CHANGELOG" already exists; SQL statement:
CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10)) [42101-220] [Failed SQL: (42101) CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10))]
        at liquibase.command.CommandScope.execute(CommandScope.java:236)
        at liquibase.Liquibase.lambda$update$0(Liquibase.java:223)
        at liquibase.Scope.lambda$child$0(Scope.java:197)
        at liquibase.Scope.child(Scope.java:206)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:175)
        at liquibase.Liquibase.runInScope(Liquibase.java:1361)
        at liquibase.Liquibase.update(Liquibase.java:215)
        at liquibase.Liquibase.update(Liquibase.java:197)
        at io.quarkus.liquibase.runtime.LiquibaseRecorder.doStartActions(LiquibaseRecorder.java:77)
        ... 13 more
Caused by: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Table "CHANGELOG" already exists; SQL statement:
CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10)) [42101-220] [Failed SQL: (42101) CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10))]
        at liquibase.executor.jvm.ChangelogJdbcMdcListener.execute(ChangelogJdbcMdcListener.java:39)
        at liquibase.changelog.StandardChangeLogHistoryService.init(StandardChangeLogHistoryService.java:277)
        at liquibase.command.core.helpers.DatabaseChangelogCommandStep.checkLiquibaseTables(DatabaseChangelogCommandStep.java:127)
        at liquibase.command.core.helpers.DatabaseChangelogCommandStep.run(DatabaseChangelogCommandStep.java:94)
        at liquibase.command.CommandScope.execute(CommandScope.java:213)
        ... 22 more
Caused by: liquibase.exception.DatabaseException: Table "CHANGELOG" already exists; SQL statement:
CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10)) [42101-220] [Failed SQL: (42101) CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10))]
        at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:467)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:180)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:145)
        at liquibase.changelog.StandardChangeLogHistoryService.lambda$init$2(StandardChangeLogHistoryService.java:277)
        at liquibase.executor.jvm.ChangelogJdbcMdcListener.lambda$execute$0(ChangelogJdbcMdcListener.java:33)
        at liquibase.Scope.lambda$child$0(Scope.java:197)
        at liquibase.Scope.child(Scope.java:206)
        at liquibase.Scope.child(Scope.java:196)
        at liquibase.Scope.child(Scope.java:175)
        at liquibase.executor.jvm.ChangelogJdbcMdcListener.execute(ChangelogJdbcMdcListener.java:32)
        ... 26 more
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "CHANGELOG" already exists; SQL statement:
CREATE TABLE PUBLIC.changelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10)) [42101-220]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
        at org.h2.message.DbException.get(DbException.java:223)
        at org.h2.message.DbException.get(DbException.java:199)
        at org.h2.command.ddl.CreateTable.update(CreateTable.java:91)
        at org.h2.command.CommandContainer.update(CommandContainer.java:169)
        at org.h2.command.Command.executeUpdate(Command.java:252)
        at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:252)
        at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:223)
        at io.agroal.pool.wrapper.StatementWrapper.execute(StatementWrapper.java:235)
        at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:461)
        ... 36 more
```

### How to Reproduce?

Reproducer in the form of a Maven project: [liquibase-issue.zip](https://github.com/quarkusio/quarkus/files/12882830/liquibase-issue.zip)

1. Build and run the attached reproducer: it uses Quarkus 3.3.3 and it should start without any issue.
2. Change the `quarkus.version` property in the `pom.xml` to `3.4.2` (currently the latest Quarkus release).
3. Rebuild and re-run the reproducer. It should terminate with a stack trace as the one above.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "21" 2023-09-19

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.4.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.4 (dfbb324ad4a7c8fb0bf182e6d91b0ae20e3d2dd9)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by appiepollo14**
Fixes #36446

### CODE REVIEW

**Comment by appiepollo14** [comment]
Does Liquibase 4.23 need to be excluded somewhere?

**Comment by geoand** [comment]
If CI is green, we can merge and backport to 3.5.

For 3.4, I'll leave it up to @gsmet to decide

**Comment by appiepollo14** [comment]
Can you explain how to do the backporting? Create a p.r. to tag 3.5?

**Comment by geoand** [comment]
You don't need to do anything, we'll take care of it.

File: bom/application/pom.xml
Comment (gsmet) [inline bom/application/pom.xml:173]:
```suggestion
        <liquibase-mongodb.version>4.24.0</liquibase-mongodb.version>
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks but could you please also update liquibase-mongodb?

**Comment by appiepollo14** [comment]
@gsmet will look into a testcase

**Comment by famod** [comment]
I think we should now be able to remove that LabelExpression workaround I had to add to the previous Liquibase update PR.
IMO, that doesn't block this PR from being merged though.

PS: I'm on PTO for the rest of this month. So if anyone else wants to remove the workaround please feel free to do so.

**Comment by appiepollo14** [comment]
Build seems to be failing on other parts of the project...

**Review by gsmet — APPROVED** [review]
I squashed the commits.

Let's see what CI has to say. It might not be fully green but it looks good enough so it should be mergeable.

**Comment by appiepollo14** [comment]
Dependabot has opened the same PR on the meantime. I haven’t found the time looking for an IT but will focus on that!

**Comment by gsmet** [comment]
@appiepollo14 I merged this but still interested if you can come up with an IT, just open a new PR. Thanks!

**Pull request merged**
